django/db/backends/oracle/introspection.py

```diff
@@ -367,6 +367,8 @@ def get_constraints(self, cursor, table_name):
                     FROM user_constraints cons
                     WHERE ind.index_name = cons.index_name
                 ) AND cols.index_name = ind.index_name
+                -- Ignore hidden columns.
+                AND NOT REGEXP_LIKE(cols.column_name, '^SYS_NC[0-9]+\$$', 'i')
```

Comment:
Unfortunately we cannot filter out hidden columns in the SQL because it will remove all functional indexes. Moreover we cannot remove them from `columns` at all because the number of columns has to match the number of orders, etc. It seems that we need to fix this directly in the `inspectdb`, e.g.
```diff
diff --git a/django/core/management/commands/inspectdb.py b/django/core/management/commands/inspectdb.py
index b48f6fcdd9..43674dccf3 100644
--- a/django/core/management/commands/inspectdb.py
+++ b/django/core/management/commands/inspectdb.py
@@ -350,7 +350,9 @@ class Command(BaseCommand):
                 columns = params["columns"]
                 if None in columns:
                     has_unsupported_constraint = True
-                columns = [x for x in columns if x is not None]
+                columns = [
+                    x for x in columns if x is not None and x in column_to_field_name
+                ]
                 if len(columns) > 1:
                     unique_together.append(
                         str(tuple(column_to_field_name[c] for c in columns))
```
What do you think?

Reply:
I was wondering if that would be the case. I also tested out a few different solutions that would keep each index, but that led `columns` and `orders` to be empty and I wasn’t sure if that would cause issues. I can implement a solution like this if you want. Thoughts?

I also like your proposed fix, it’s simple and the bug is an issue with `inspectdb` in the first place.

Reply:
> but that led columns and orders to be empty and I wasn’t sure if that would cause issues. I can implement a solution like this if you want. Thoughts?

I tried this locally. Unfortunately, this causes functional indexes tests where we inspect columns ordering to fail.

Reply:
I can go ahead and implement your solution in `inspectdb`. Another issue I found with removing items from the `columns` is that it incorrectly marked some columns as unique when creating `unique_columns`.

Reply:
Tested the fix against my database and it works properly.

Reply:
Thanks :+1: